realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java

```diff
@@ -605,9 +611,17 @@ public Builder errorHandler(SyncSession.ErrorHandler errorHandler) {
          * <p>
          * This check is only enforced the first time a Realm is created. If you otherwise want to make sure a Realm
          * has the latest changes, use {@link SyncSession#downloadAllServerChanges()}.
+         *
+         * @param readOnly if {@code true} no local changes can be made to the Realm once downloaded. All write
+         * transactions will fail with an {@link IllegalStateException}. Marking a Realm as read-only only applies to
+         * the Realm in this process. Other processes or devices can still write to the Realm.
          */
-        public Builder waitForInitialRemoteData() {
+        public Builder waitForInitialRemoteData(boolean readOnly) {
```

Comment:
Not entirely sure what you mean? This setting is only available on the SyncConfiguration Builder which only works for synced Realms?